legion/engine/physics/systems/physicssystem.cpp

```diff
@@ -201,12 +126,21 @@ namespace legion::physics
         //the effective mass remains the same for every iteration of the solver. This means that we can precalculate it before
         //we start the solver
 
-        //log::debug("--------------Logging contacts for manifold -------------------");
         {
             OPTICK_EVENT("Resolve collisions");
 
             initializeManifolds(manifoldsToSolve, manifoldValidity);
 
+            auto largestPenetration = [](const physics_contact& contact1, const physics_contact& contact2)
+            -> bool {
+            return math::dot(contact1.RefWorldContact - contact1.IncWorldContact, -contact1.collisionNormal)
+            > math::dot(contact2.RefWorldContact - contact2.IncWorldContact, -contact2.collisionNormal);};
+
+            for (auto& manifold : manifoldsToSolve)
+            {
+                std::sort(manifold.contacts.begin(), manifold.contacts.end(), largestPenetration);
+            }
```

Comment:
personally I'm not a big fan of unnecessary lambdas that also aren't in-place. I can argue for single use lambdas in an external api. but in internal code it tends to make little sense to make a `for_each` function to make the code more declarative and then whenever you use it making your code just as imperative as it would've otherwise been by using the lambda. The only reason I can see to do so is if the `for_each` abstracts away considerable amounts of logic.

in this case I think it would've been fine if the lambda was in-place in the sort function, and written a bit more readably by doing something like:
```cpp
auto dot1 = math::dot(contact1.RefWorldContact - contact1.IncWorldContact, -contact1.collisionNormal);
auto dot2 = math::dot(contact2.RefWorldContact - contact2.IncWorldContact, -contact2.collisionNormal);
return dot1 > dot2;
```

but preferably i'd take the logic out of this already big function by declaring it somewhere else in the cpp file like this:
```cpp
namespace detail
{
    static inline bool greater_penetration(const physics_contact& contact1, const physics_contact& contact2)
    {
        auto dot1 = math::dot(contact1.RefWorldContact - contact1.IncWorldContact, -contact1.collisionNormal);
        auto dot2 = math::dot(contact2.RefWorldContact - contact2.IncWorldContact, -contact2.collisionNormal);
        return dot1 > dot2;
    }
}
...
for (auto& manifold : manifoldsToSolve)
{
    std::sort(manifold.contacts.begin(), manifold.contacts.end(), &detail::greater_penetration);
}
```
cuz if you're gonna write more declarative code instead, might as well at least go all the way.